### PR TITLE
CLOUDP-167800: Migrate Custom DNS to v2 client

### PR DESCRIPTION
**Pull request by andmatei**
## Proposed changes
Migrate Custom DNS methods to v2 client.

_Jira ticket:_ CLOUDP-167800

Closes #[issue number]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/atlas/customdns/aws/describe_test.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
 	"github.com/mongodb/mongodb-atlas-cli/internal/mocks"
 	"github.com/mongodb/mongodb-atlas-cli/internal/test"
-	atlas "go.mongodb.org/atlas/mongodbatlas"
+	atlas "go.mongodb.org/atlas/mongodbatlasv2"
```
Comment (wtrocki) [inline internal/cli/atlas/customdns/aws/describe_test.go:26]: super nit
IMHO good pattern is to use atlasv2 or v2 to not confuse us about what is migtated. Same name across the board.
@swattyT @matt-condon

**Comment by wtrocki** [comment]
It will be good to get info what type of testing were done locally.
Verification will be much easier.

**Review by wtrocki — APPROVED** [review]
LGTM.

**Review by blva — APPROVED** [review]
🚀

File: internal/cli/atlas/customdns/aws/describe_test.go (hunk 1 above)
Comment (blva) [reply internal/cli/atlas/customdns/aws/describe_test.go:26]: [nit] based on https://github.com/mongodb/mongodb-atlas-cli/blob/4e11a40faf0d02a90228f84a303809f2de7a361f/internal/store/access_logs.go it seems like in some places both `mongodbatlas` and `mongodbatlasv2` will exist in the same file. should we stick to`atlasv2` here? ideally could we agree on same naming for all PRs and files?

Comment (andmatei) [reply internal/cli/atlas/customdns/aws/describe_test.go]: Done, renamed all instances of the v2 client to `atlasv2`

**Pull request merged**